src/NUnitFramework/framework/Api/NUnitTestAssemblyRunner.cs

```diff
@@ -339,15 +339,16 @@ private void OnRunCompleted(object sender, EventArgs e)
             _runComplete.Set();
         }
 
-        private int CountTestCases(ITest test, ITestFilter filter)
+        private static int CountTestCases(ITest test, ITestFilter filter)
         {
             if (!test.IsSuite)
                 return filter.Pass(test) ? 1: 0;
 
             int count = 0;
-            foreach (ITest child in test.Tests)
+            var tests = test.Tests;
+            for (var i = 0; i < tests.Count; i++)
```

Comment:
`for` loop against IList is the best choice as no need for boxed enumerator (IList source isn't known and direct struct enumerator from `GetEnumerator` isn't being used).